contrib/gorm.io/gorm.v1/option.go

```diff
@@ -71,3 +74,19 @@ func WithErrorCheck(fn func(err error) bool) Option {
 		cfg.errCheck = fn
 	}
 }
+
+// WithCustomTag will cause the given tagFn to be evaluated after executing
+// a query and attach the result to the span tagged by the key.
+func WithCustomTag(tag string, tagFn func(db *gorm.DB) interface{}) Option {
+	return func(cfg *config) {
+		if cfg.tagFns == nil {
+			cfg.tagFns = make(map[string]func(scope *gorm.Scope) interface{})
+		}
+
+		if tagFn != nil {
+			cfg.tagFns[tag] = tagFn
+		} else {
+			delete(cfg.tagFns, tag)
```

Comment:
I'm curious why you went with a slightly different approach here compared to the other `WithCustomTag` function? Do you foresee needing to be able to remove/delete tagFns after they've already been added?

Reply:
tbh I don't think it's practically useful. added this implementation to behave in a way that callers would assume. Not a must-have and I can remove it if you prefer.